src/prefect/utilities/executors.py

```diff
@@ -174,11 +174,42 @@ def multiprocessing_safe_run_and_retrieve(
         with prefect.context(context):
             logger.debug(f"{name}: Executing...")
             return_val = fn(*args, **kwargs)
+            logger.debug(f"{name}: Execution successful.")
     except Exception as exc:
         return_val = exc
+        logger.error(
+            f"{name}: Encountered a {type(exc).__name__}, "
+            f"returning details as a result..."
+        )
+
+    try:
+        pickled_val = cloudpickle.dumps(return_val)
+    except Exception as exc:
+        base_msg = (
+            f"Failed to pickle result of type {type(return_val).__name__!r} with "
+            f'exception: "{type(exc).__name__}: {str(exc)}".'
+        )
+        logger.error(f"{name}: {base_msg}")
+        pickled_val = cloudpickle.dumps(
+            RuntimeError(
+                f"{base_msg} This timeout handler requires your function return "
+                f"value to be serializable with `cloudpickle`. "
+                "If you must return a unserializable value, consider switching your "
+                "executor to use processes instead of threads to use a different "
+                "timeout handler."
+            )
```

Comment:
Now that I think about it more -- won't dask fail with these kind of results anyway? I guess you could get away with using the `LocalExecutor` ? hmm.

Reply:
I've removed this extra message because I do not think it will help.